actors/miner/tests/util.rs

```diff
@@ -204,8 +261,1024 @@ impl ActorHarness {
+            penalty_from_vesting -= cfg.repaid_fee_debt.clone();
+            // Precommit deposit burns are repaid from PCD account
+            penalty_from_vesting -= cfg.expired_precommit_penalty.clone();
+            // New penalties are paid first from vesting funds but, if exhausted, overflow to unlocked balance.
+            penalty_from_vesting -= cfg.penalty_from_unlocked.clone();
+
+            pledge_delta -= penalty_from_vesting;
+        }
+
+        pledge_delta += cfg.expired_sectors_pledge_delta;
+        pledge_delta -= immediately_vesting_funds(rt, &state);
+
+        if pledge_delta != TokenAmount::from(0) {
+            rt.expect_send(
+                *STORAGE_POWER_ACTOR_ADDR,
+                PowerMethod::UpdatePledgeTotal as u64,
+                RawBytes::serialize(BigIntSer(&pledge_delta)).unwrap(),
+                TokenAmount::from(0),
+                RawBytes::default(),
+                ExitCode::Ok,
+            );
+        }
+
+        // Re-enrollment for next period.
+        if !cfg.no_enrollment {
+            let params = make_deadline_cron_event_params(cfg.expected_enrollment);
+            rt.expect_send(
+                *STORAGE_POWER_ACTOR_ADDR,
+                PowerMethod::EnrollCronEvent as u64,
+                RawBytes::serialize(params).unwrap(),
+                TokenAmount::from(0),
+                RawBytes::default(),
+                ExitCode::Ok,
+            );
+        }
+
+        let params = make_deferred_cron_event_params(
+            self.epoch_reward_smooth.clone(),
+            self.epoch_qa_power_smooth.clone(),
+        );
+        rt.set_caller(*POWER_ACTOR_CODE_ID, *STORAGE_POWER_ACTOR_ADDR);
+        rt.call::<Actor>(Method::OnDeferredCronEvent as u64, &RawBytes::serialize(params).unwrap())
+            .unwrap();
+        rt.verify();
+    }
+
+    pub fn submit_window_post(
+        &self,
+        rt: &mut MockRuntime,
+        deadline: &DeadlineInfo,
+        partitions: Vec<PoStPartition>,
+        infos: Vec<SectorOnChainInfo>,
+        cfg: PoStConfig,
+    ) {
+        let params = SubmitWindowedPoStParams {
+            deadline: deadline.index,
+            partitions: partitions,
+            proofs: make_post_proofs(self.window_post_proof_type),
+            chain_commit_epoch: deadline.challenge,
+            chain_commit_rand: Randomness(Vec::from(b"chaincommitment".clone())),
+        };
+        self.submit_window_post_raw(rt, deadline, infos, params, cfg).unwrap();
+        rt.verify();
+    }
+
+    fn submit_window_post_raw(
+        &self,
+        rt: &mut MockRuntime,
+        deadline: &DeadlineInfo,
+        infos: Vec<SectorOnChainInfo>,
+        params: SubmitWindowedPoStParams,
+        cfg: PoStConfig,
+    ) -> Result<RawBytes, ActorError> {
+        rt.set_caller(*ACCOUNT_ACTOR_CODE_ID, self.worker);
+        let chain_commit_rand = match cfg.chain_randomness {
```

Comment:
Pathological case but I think this has different behavior than go if `cfg.chain_randomness` is `Some(Vec::from(b""))`

Reply:
sure, but why would we do this? 
The whole point of using an option is to pass None if we dont have any.